src/ESPAsyncWebServer.h

```diff
@@ -408,6 +409,55 @@ class AsyncWebServerRequest {
     const AsyncWebHeader* getHeader(const __FlashStringHelper* data) const;
 #endif
     const AsyncWebHeader* getHeader(size_t num) const;
+    size_t getHeaderNames(std::vector<const char*>& names) const {
+      names.clear();
+      const size_t size = _headers.size();
+      names.reserve(size);
+      for (const auto& h : _headers) {
+        names.push_back(h.name().c_str());
+      }
+      return size;
+    }
+    const std::list<AsyncWebHeader>& getHeaders() { return _headers; }
```

Comment:
This method must be cons also?  `const std::list<AsyncWebHeader>& getHeaders() const { return _headers; }`